includes/api/class-wc-rest-data-controller.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * REST API Data controller
+ *
+ * Handles requests to the /data/location endpoint.
+ *
+ * @author   WooThemes
+ * @category API
+ * @package  WooCommerce/API
+ * @since    2.6.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * REST API Data controller class.
+ *
+ * @package WooCommerce/API
+ * @extends WC_REST_Controller
+ */
+class WC_REST_Data_Index_Controller extends WC_REST_Data_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v2';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'data';
+
+	/**
+	 * Register routes.
+	 *
+	 * @since 3.1.0
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+		) );
+	}
+
+	/**
+	 * Return the list of data resources
+	 *
+	 * @since  3.1.0
+	 * @param  WP_REST_Request $request
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$data    = array();
+		$resources = array(
+			array(
+				'slug' => 'locations',
+				'description' => __( 'List of supported continents, countries, and states', 'woocommerce' ),
+			),
+			array(
+				'slug' => 'locations/<continent>',
+				'description' => __( 'List of supported continents, countries, and states; restricted to the given continent', 'woocommerce' ),
+			),
+			array(
+				'slug' => 'locations/<continent>/<country>',
```

Comment:
Is having just continent useful? We could just get rid of `/data/locations/<continent>` and only have `/data/locations/<country>`.